plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -25,34 +30,63 @@ import org.jetbrains.kotlin.formver.viper.domains.CastingDomain
  * We need the FirSession to get access to the TypeContext.
  */
 class ProgramConverter(val session: FirSession) : ProgramConversionContext {
+
     private val methods: MutableMap<MangledName, Method> = mutableMapOf()
+    private val classes: MutableMap<ClassName, ClassEmbedding> = mutableMapOf()
 
     val program: Program
         get() = Program(
-            listOf(UnitDomain, NullableDomain, CastingDomain), /* Domains */
-            SpecialFields.all, /* Fields */
-            SpecialMethods.all + methods.values.toList(), /* Methods */
+            domains = listOf(UnitDomain, NullableDomain, CastingDomain), /* Domains */
+            fields = SpecialFields.all + classes.values.flatMap { it.fields }.map { it.toField() }, /* Fields */
+            methods = SpecialMethods.all + methods.values.toList(), /* Methods */
         )
 
     fun addWithBody(declaration: FirSimpleFunction) {
         processFunction(declaration.symbol, declaration.body)
     }
 
-    override fun add(symbol: FirNamedFunctionSymbol): MethodSignatureEmbedding {
+    override fun add(symbol: FirFunctionSymbol<*>): MethodSignatureEmbedding {
         return processFunction(symbol, null)
     }
 
+    override fun add(symbol: FirRegularClassSymbol): ClassEmbedding {
+
+        val className = ClassName(symbol.classId.packageFqName, symbol.classId.shortClassName)
+        // If the class name is not contained in the classes hashmap, then add a new embedding.
+        return classes.getOrPut(className) l@{
+            // Get classes fields
+            val concreteFields = symbol.declarationSymbols
+                .filterIsInstance<FirPropertySymbol>()
+                .filter { it.hasBackingField }
+                .map { VariableEmbedding(it.callableId.embedName(), embedType(it.resolvedReturnType)) }
+                .toMutableList()
+
+            return@l ClassEmbedding(className, concreteFields)
```

Comment:
Nit: You don't need the label here `ClassEmbedding(className, concreteFields)` without the return can just be the last expression.